--- ks_bmtests.cfg.orig	2014-09-10 13:06:43.524253992 -0400
+++ ks_bmtests.cfg	2014-09-11 11:30:54.412254138 -0400
@@ -342,6 +342,9 @@
     fi
 }
 
+echo -e "DEVICE=eth0:1\nBOOTPROTO=static\nONBOOT=yes\nIPADDR=10.16.122.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth0:1
+echo -e "DEVICE=eth1\nBOOTPROTO=static\nONBOOT=yes\nIPADDR=192.168.50.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth1
+
 # Default FQDN
 hostname="nailgun.mirantis.com"
 
@@ -433,6 +436,12 @@
 chmod 0777 /usr/local/sbin/bootstrap_admin_node.sh
 cp ${SOURCE}/bootstrap_admin_node.conf /etc/init/bootstrap_admin_node.conf
 echo "ENABLED=1" > /etc/sysconfig/bootstrap_admin_node
+# Dirty hacks for bare-metal automation
+sed -e 's#iface=eth0#iface=eth1#' -i /usr/local/sbin/bootstrap_admin_node.sh
+cat > /var/log/docker-logs/cobbler/anaconda_lvm_bug_hack.sh << EOF
+echo -e "#raw\nwhile : ; do\n\tegrep -q 'CRITICAL: anaconda .* exception report' /tmp/anaconda.log && egrep -q 'LVMError: lvactivate failed' /tmp/anaconda.log && reboot ;\n\tsleep 10;\ndone &\n#end raw" > /var/lib/cobbler/snippets/anaconda_lvm_bug_workaround
+sed -e "s#('pre_install_network_config')#('pre_install_network_config')\n\n\\\$SNIPPET('anaconda_lvm_bug_workaround')#" -i /var/lib/cobbler/kickstarts/centos-x86_64.ks
+EOF
 
 # Copying version.yaml file. It contains COMMIT_SHA of last commit.
 RELEASE=$(awk '/release/{gsub(/"/, "");print $2}' ${SOURCE}/version.yaml)
@@ -498,6 +507,8 @@
 
 EOF
 
+echo >> /etc/rc.local
+echo "iptables -t nat -I POSTROUTING -s 10.16.122.0/24 -j MASQUERADE" >> /etc/rc.local
 
 # Unmounting source
 umount -f ${SOURCE}
@@ -531,4 +542,4 @@
 # Blacklist i2c_piix4 module for VirtualBox so it does not create kernel errors
 [[ $(virt-what) = "virtualbox" ]] && echo "blacklist i2c_piix4" > /etc/modprobe.d/blacklist-i2c-piix4.conf
 
-%end
\ No newline at end of file
+%end
