--- /var/lib/tftpboot/fuel_bmtests/ks.cfg	2014-03-14 06:09:51.000000000 -0400
+++ /var/lib/tftpboot/ks_bmtests.cfg	2014-03-17 04:17:52.472254353 -0400
@@ -29,7 +29,7 @@
     fi
 done
 
-installdrive="undefined"
+installdrive="sda"
 for I in `cat /proc/cmdline`; do case "$I" in *=*) eval $I;; esac ; done
 
 set ${drives}
@@ -246,6 +246,9 @@
     fi
 }
 
+echo -e "DEVICE=eth0:1\nBOOTPROTO=static\nIPADDR=10.16.122.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth0:1
+echo -e "DEVICE=eth1\nBOOTPROTO=static\nIPADDR=192.168.50.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth1
+
 # Default FQDN
 hostname="nailgun.mirantis.com"
 
@@ -337,6 +340,12 @@
 chmod 0777 /usr/local/sbin/bootstrap_admin_node.sh
 cp ${SOURCE}/bootstrap_admin_node.conf /etc/init/bootstrap_admin_node.conf
 echo "ENABLED=1" > /etc/sysconfig/bootstrap_admin_node
+# Dirty hacks for bare-metal automation
+sed -e 's#iface=eth0#iface=eth1#' -i /usr/local/sbin/bootstrap_admin_node.sh
+cat > /usr/local/sbin/anaconda_lvm_bug_hack.sh << EOF
+echo -e "#raw\nwhile : ; do\n\tegrep -q 'CRITICAL: anaconda .* exception report' /tmp/anaconda.log && egrep -q 'LVMError: lvactivate failed' /tmp/anaconda.log && reboot ;\n\tsleep 10;\ndone &\n#end raw" > /var/lib/cobbler/snippets/anaconda_lvm_bug_workaround
+sed -e "s#('pre_install_network_config')#('pre_install_network_config')\n\n\\\$SNIPPET('anaconda_lvm_bug_workaround')#" -i /var/lib/cobbler/kickstarts/centos-x86_64.ks
+EOF
 
 # Copying version.yaml file. It contains COMMIT_SHA of last commit.
 mkdir -p /etc/nailgun
@@ -370,6 +379,8 @@
 EOF
 
 echo "sed -i \"s%\(^.*able on:\).*$%\1 http://\`ip address show eth0 | awk '/inet / {print \$2}' | cut -d/ -f1 -\`:8000%\" /etc/issue" >>/etc/rc.local
+echo >> /etc/rc.local
+echo "iptables -t nat -I POSTROUTING -s 10.16.122.0/24 -j MASQUERADE" >> /etc/rc.local
 
 # Unmounting source
 umount -f ${SOURCE}
