--- fuel_bmtests/ks.cfg	2014-03-04 03:50:41.000000000 -0500
+++ ks_bmtests.cfg	2014-03-04 11:51:34.188252527 -0500
@@ -31,7 +31,7 @@
 
 set ${drives}
 numdrives=`echo $#`
-tgtdrive="undefined"
+tgtdrive="sda"
 if [ ${numdrives} -gt 1 ]; then
     exec < /dev/tty3 > /dev/tty3 2>&1
     chvt 3
@@ -241,6 +241,9 @@
     fi
 }
 
+echo -e "DEVICE=eth0:1\nBOOTPROTO=static\nIPADDR=10.16.122.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth0:1
+echo -e "DEVICE=eth1\nBOOTPROTO=static\nIPADDR=192.168.50.1\nNETMASK=255.255.255.0\n" > /etc/sysconfig/network-scripts/ifcfg-eth1
+
 # Default FQDN
 hostname="nailgun.mirantis.com"
 
@@ -332,6 +335,12 @@
 chmod 0777 /usr/local/sbin/bootstrap_admin_node.sh
 cp ${SOURCE}/bootstrap_admin_node.conf /etc/init/bootstrap_admin_node.conf
 echo "ENABLED=1" > /etc/sysconfig/bootstrap_admin_node
+# Dirty hacks for bare-metal automation
+sed -e 's#iface=eth0#iface=eth1#' -i /usr/local/sbin/bootstrap_admin_node.sh
+cat > /usr/local/sbin/anaconda_lvm_bug_hack.sh << EOF
+echo -e "#raw\nwhile : ; do\n\tegrep -q 'CRITICAL: anaconda .* exception report' /tmp/anaconda.log && egrep -q 'LVMError: lvactivate failed' /tmp/anaconda.log && reboot ;\n\tsleep 10;\ndone &\n#end raw" > /var/lib/cobbler/snippets/anaconda_lvm_bug_workaround
+sed -e "s#('pre_install_network_config')#('pre_install_network_config')\n\n\\\$SNIPPET('anaconda_lvm_bug_workaround')#" -i /var/lib/cobbler/kickstarts/centos-x86_64.ks
+EOF
 
 # Copying version.yaml file. It contains COMMIT_SHA of last commit.
 mkdir -p /etc/nailgun
@@ -365,6 +374,8 @@
 EOF
 
 echo "sed -i \"s%\(^.*able on:\).*$%\1 http://\`ip address show eth0 | awk '/inet / {print \$2}' | cut -d/ -f1 -\`:8000%\" /etc/issue" >>/etc/rc.local
+echo >> /etc/rc.local
+echo "iptables -t nat -I POSTROUTING -s 10.16.122.0/24 -j MASQUERADE" >> /etc/rc.local
 
 # Unmounting source
 umount -f ${SOURCE}
@@ -401,4 +412,4 @@
 # Blacklist i2c_piix4 module for VirtualBox so it does not create kernel errors
 [[ $(virt-what) = "virtualbox" ]] && echo "blacklist i2c_piix4" > /etc/modprobe.d/blacklist-i2c-piix4.conf
 
-%end
\ No newline at end of file
+%end
